packages/assets-controllers/src/AssetsContractController.ts

```diff
@@ -348,21 +429,17 @@ export class AssetsContractController extends BaseController<
    * @param userAddress - Wallet public address.
    * @param nftAddress - ERC1155 asset contract address.
    * @param nftId - ERC1155 asset identifier.
+   * @param networkClientId - Network Client ID to fetch the provider with.
    * @returns Promise resolving to the 'balanceOf'.
    */
   async getERC1155BalanceOf(
     userAddress: string,
     nftAddress: string,
     nftId: string,
+    networkClientId?: NetworkClientId,
   ): Promise<BN> {
-    if (this.erc1155Standard === undefined) {
-      throw new Error(MISSING_PROVIDER_ERROR);
-    }
-    return await this.erc1155Standard.getBalanceOf(
-      nftAddress,
-      userAddress,
-      nftId,
-    );
+    const erc1155Standard = this.getERC1155Standard(networkClientId);
+    return erc1155Standard.getBalanceOf(nftAddress, userAddress, nftId);
```

Comment:
Just calling out that we remove the `await` here. Doesn't make any difference. just noting it.

Reply:
yeah, unnecessary await that i decided to drop since I was touching the line anyway